### ISSUE TITLE
fails `cross check --target x86_64-pc-solaris` on 1.72.1

### ISSUE THREAD

**Issue by jtmoon79** [bytecodealliance/rustix#1217]
[Cross-compiling `rustix` 0.38.37 fails for platform `x86_64-pc-solaris`](https://github.com/jtmoon79/super-speedy-syslog-searcher/actions/runs/11873370669/job/33088378752).

Running on Ubuntu 22, using rust 1.72.1
```
$ cross --version
cross 0.2.5
cargo 1.72.1 (103a7ff2e 2023-08-15)

$ cross check --locked --lib --bins --target x86_64-pc-solaris
...
   Compiling syn v2.0.66
    Checking errno v0.3.8
    Checking memchr v2.7.4
    Checking encoding_index_tests v0.1.4
    Checking once_cell v1.20.2
error[E0425]: cannot find value `LOCK_SH` in module `c`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.37/src/backend/libc/fs/types.rs:950:30
    |
950 |     LockShared = bitcast!(c::LOCK_SH),
    |                              ^^^^^^^ not found in `c`

error[E0425]: cannot find value `LOCK_EX` in module `c`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.37/src/backend/libc/fs/types.rs:952:33
    |
952 |     LockExclusive = bitcast!(c::LOCK_EX),
    |                                 ^^^^^^^ not found in `c`

error[E0425]: cannot find value `LOCK_UN` in module `c`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.37/src/backend/libc/fs/types.rs:954:26
    |
954 |     Unlock = bitcast!(c::LOCK_UN),
    |                          ^^^^^^^ not found in `c`

error[E0425]: cannot find value `LOCK_SH` in module `c`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.37/src/backend/libc/fs/types.rs:956:41
    |
956 |     NonBlockingLockShared = bitcast!(c::LOCK_SH | c::LOCK_NB),
    |                                         ^^^^^^^ not found in `c`

error[E0425]: cannot find value `LOCK_NB` in module `c`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.37/src/backend/libc/fs/types.rs:956:54
    |
956 |     NonBlockingLockShared = bitcast!(c::LOCK_SH | c::LOCK_NB),
    |                                                      ^^^^^^^ not found in `c`

error[E0425]: cannot find value `LOCK_EX` in module `c`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.37/src/backend/libc/fs/types.rs:958:44
    |
958 |     NonBlockingLockExclusive = bitcast!(c::LOCK_EX | c::LOCK_NB),
    |                                            ^^^^^^^ not found in `c`

error[E0425]: cannot find value `LOCK_NB` in module `c`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.37/src/backend/libc/fs/types.rs:958:57
    |
958 |     NonBlockingLockExclusive = bitcast!(c::LOCK_EX | c::LOCK_NB),
    |                                                         ^^^^^^^ not found in `c`

error[E0425]: cannot find value `LOCK_UN` in module `c`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.37/src/backend/libc/fs/types.rs:960:37
    |
960 |     NonBlockingUnlock = bitcast!(c::LOCK_UN | c::LOCK_NB),
    |                                     ^^^^^^^ not found in `c`

error[E0425]: cannot find value `LOCK_NB` in module `c`
   --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.37/src/backend/libc/fs/types.rs:960:50
    |
960 |     NonBlockingUnlock = bitcast!(c::LOCK_UN | c::LOCK_NB),
    |                                                  ^^^^^^^ not found in `c`

error[E0425]: cannot find value `VSTATUS` in module `c`
    --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.37/src/termios/types.rs:1196:39
     |
1196 |     pub const VSTATUS: Self = Self(c::VSTATUS as usize);
     |                                       ^^^^^^^ help: a constant with a similar name exists: `P_STATUS`
     |
    ::: /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libc-0.2.164/src/unix/solarish/mod.rs:2449:1
     |
2449 | pub const P_STATUS: ::c_int = 0x003;
     | --------------------------- similarly named constant `P_STATUS` defined here

error[E0425]: cannot find value `VERASE2` in module `c`
    --> /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.38.37/src/termios/types.rs:1200:39
     |
1200 |     pub const VERASE2: Self = Self(c::VERASE2 as usize);
     |                                       ^^^^^^^ help: a constant with a similar name exists: `VERASE`
     |
    ::: /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libc-0.2.164/src/unix/solarish/mod.rs:2280:1
     |
2280 | pub const VERASE: usize = 2;
     | ----------------------- similarly named constant `VERASE` defined here
...
```

I found this by way of updating dependency `libc` from 0.2.159 to 0.2.164 which compiles `rustix` 0.38.37.

I have not tried other rust versions or platform configurations.

**Comment by jtmoon79** [comment]
Closing. I'm unable to reproduce this error when compiling _only_ the `rustix` project.

```
$ git clone https://github.com/bytecodealliance/rustix.git
$ cd rustix
$ rustup override set 1.72.1
$ cross --version
cross 0.2.5
[cross] note: Falling back to `cargo` on the host.
cargo 1.72.1 (103a7ff2e 2023-08-15)
$ cross check --locked --lib --bins --target x86_64-pc-solaris
```

**Comment by sunfishcode** [comment]
These errors were fixed in rustix 0.38.40.
